nengo_spinnaker/regions/profiler.py

```diff
@@ -0,0 +1,95 @@
+import logging
+import numpy as np
+import struct
+
+from .region import Region
+
+MS_SCALE = (1.0 / 200032.4)
+
+logger = logging.getLogger(__name__)
+
+
+class Profiler(Region):
+    """Region used to record spikes."""
+    def __init__(self, n_samples):
+        self.n_samples = n_samples
+
+    def sizeof(self, vertex_slice):
+        # 1 word counter and 2 words per sample
+        return 4 + (8 * self.n_samples)
+
+    def write_subregion_to_file(self, *args, **kwargs):  # pragma: no cover
+        pass  # Nothing to do
+
+    def read_from_mem(self, mem, tag_names):
+        # Read number of words written by profiler
+        word_written = struct.unpack("I", mem.read(4))[0]
+
+        # Read these from memory
+        data = np.fromstring(mem.read(word_written * 4), dtype=np.uint64)
```

Comment:
Yes it is - the profiler writes out pairs of uint32s

On 10 August 2015 at 11:12, Andrew Mundy <notifications@github.com> wrote:

> In nengo_spinnaker/regions/profiler.py
> <https://github.com/project-rig/nengo_spinnaker/pull/52#discussion_r36617388>
> :
>
> > +    def __init__(self, n_samples):
> > +        self.n_samples = n_samples
> > +
> > +    def sizeof(self, vertex_slice):
> > +        # 1 word counter and 2 words per sample
> > +        return 4 + (8 * self.n_samples)
> > +
> > +    def write_subregion_to_file(self, *args, **kwargs):  # pragma: no cover
> > +        pass  # Nothing to do
> > +
> > +    def read_from_mem(self, mem, tag_names):
> > +        # Read number of words written by profiler
> > +        word_written = struct.unpack("I", mem.read(4))[0]
> > +
> > +        # Read these from memory
> > +        data = np.fromstring(mem.read(word_written * 4), dtype=np.uint64)
>
> I'm guessing that word_written is always even?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/project-rig/nengo_spinnaker/pull/52/files#r36617388>.
>